libgalois/src/PropertyGraph.cpp

```diff
@@ -413,6 +414,16 @@ katana::PropertyGraph::Make(katana::GraphTopology&& topo_to_assign) {
   return std::make_unique<katana::PropertyGraph>(std::move(topo_to_assign));
 }
 
+katana::Result<std::unique_ptr<katana::PropertyGraph>>
+katana::PropertyGraph::Make(
+    katana::GraphTopology&& topo_to_assign,
+    NUMAArray<EntityTypeID> node_entity_type_id,
+    NUMAArray<EntityTypeID> edge_entity_type_id) {
+  return std::make_unique<katana::PropertyGraph>(
+      std::move(topo_to_assign), std::move(node_entity_type_id),
+      std::move(edge_entity_type_id));
```

Comment:
How does this work without the entity-type-id maps?

Reply:
I actually hadn't thought it out that far. I suppose if we are passing in these arrays at construction time, we should be passing in the maps at construction time as well.

Reply:
Yes, otherwise, there's no way to decouple it. I'm working on moving the "maps" to a single class. But for now, you can pass each of those during construction.

Reply:
And then we will have to do the "infer all maps from the minimal set of maps" step here as well.

Reply:
No, the class object will do the inferral if needed.

Reply:
I know I pushed for a "just construct the full set of maps from the minimal set" approach, but I am changing my tune. The minimal set of maps is 4 maps and the full set is 8. I think we should wait to pass maps here until we have a sensible class wrapping them.

If we want to generate some maps from others, we should put that logic in the class that wraps the maps (looks like Roshan beat me to this one).

Reply:
Agree, PropertyGraph should keep a reference or copy of a `EntityTypeMappings` class that encapsulates the type ID arrays and maps (and the API).